python/dolfinx/wrappers/la.cpp

```diff
@@ -26,8 +26,8 @@ namespace py = pybind11;
 
 namespace
 {
-// ScatterMode types
-enum class PyScatterMode
+// InsertMode types
+enum class PyInsertMode
```

Comment:
I don't think this is clearer. What's being controlled is what happens during a scatter. Moreover, the previous naming had a consistency with PETSc, see https://petsc.org/release/manualpages/Vec/VecGhostUpdateBegin/.

The name change could be confused with switching between insert and set during assembly.

Reply:
I think it was inconsistent with PETSc before, and is more inline now: PETSc has a scattermode (forward or reverse).